src/utils/time-utils.ts

```diff
@@ -0,0 +1,107 @@
+import {
+    CAPTURE_INTERVAL_RE,
+    DURATION_RE,
+    POSTGRES_INTERVAL_RE,
+} from 'validation';
+import { hasLength } from './misc-utils';
+
+interface SegmentedInterval {
+    h: number;
+    m: number;
+    s: number;
+}
+
+const parsePostgresInterval = (interval: string): SegmentedInterval => {
+    const [hours, minutes, seconds] = interval.split(':').map((segment) => {
+        const numericSegment = Number(segment);
+
+        return isFinite(numericSegment) ? numericSegment : -1;
+    });
+
+    return { h: hours, m: minutes, s: seconds };
+};
+
+const parseCaptureInterval = (interval: string): SegmentedInterval => {
+    let hours = -1;
+    let minutes = -1;
+    let seconds = -1;
+
+    interval.split(' ').forEach((segment) => {
+        const numericSegment = Number(segment.substring(0, segment.length - 1));
+        const numericValue = isFinite(numericSegment) ? numericSegment : -1;
+
+        if (segment.includes('h')) {
+            hours = numericValue;
+        } else if (segment.includes('m')) {
+            minutes = numericValue;
+        } else if (segment.includes('s')) {
+            seconds = numericValue;
+        }
+    });
+
+    return { h: hours, m: minutes, s: seconds };
+};
+
+export const getCaptureIntervalSegment = (interval: string, unit: string) => {
+    let truncatedInterval = -1;
+
+    const granularTimeUnit = unit === 'h' || unit === 'm' || unit === 's';
+
+    if (granularTimeUnit && POSTGRES_INTERVAL_RE.test(interval)) {
+        const segments = parsePostgresInterval(interval);
+
+        truncatedInterval = segments[unit];
+    }
+
+    if (granularTimeUnit && CAPTURE_INTERVAL_RE.test(interval)) {
+        const segments = parseCaptureInterval(interval);
+
+        truncatedInterval = segments[unit];
+    }
+
+    return truncatedInterval;
+};
+
+export const formatCaptureInterval = (
+    interval: string | null | undefined,
+    intervalUnitSupported?: boolean
+): string | null => {
+    if (typeof interval !== 'string') {
+        return null;
+    }
+
+    let formattedInterval = '';
```

Comment:
Instead of using concat on a string this should just be an array that is joined at the end.

Reply:
That also helps in making sure the last element does not have the join char.